invenio_rdm_records/resources/serializers/codemeta/schema.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Codemeta schema."""
+
+from edtf import parse_edtf
+from edtf.parser.grammar import ParseException
+from marshmallow import ValidationError, fields, missing
+from pydash import py_
+
+from ..schemaorg.schema import SchemaorgSchema
+
+
+class CodemetaSchema(SchemaorgSchema):
+    """CodeMeta schema.
+
+    CodeMeta schema derives from Schema.org
+    """
+
+    context = fields.Constant(
+        "https://doi.org/10.5063/schema/codemeta-2.0", data_key="@context"
+    )
+    funding = fields.Method("get_funding")
+    embargoDate = fields.Method("get_embargo_date", data_key="embargoDate")
+
+    def get_type(self, obj):
+        """Get type.
+
+        For CodeMeta, only "SoftwareSourceCode" types are allowed.
+        """
+        read_type = super().get_type(obj)
+        if "SoftwareSourceCode" not in read_type:
+            raise ValidationError(
+                "Record is not CodeMeta serializable. It must be of type software."
+            )
```

Comment:
minor: since we're not using `CodeMeta` as input, I wouldn't raise a validation error here... I would still actually serialize with the original type, and if there's some limitation about the type, enforce it higher up the call chain.